igor-web/src/main/groovy/com/netflix/spinnaker/igor/travis/TravisBuildMonitor.groovy

```diff
@@ -92,6 +100,19 @@ class TravisBuildMonitor implements PollingMonitor{
                 }
             } as Action0, 0, pollInterval, TimeUnit.SECONDS
         )
+
+        worker.schedulePeriodically(
+            {
+                if (isInService()) {
```

Comment:
Can you be sure that these workers won't step on each other if the sync interval is too low? Maybe protect the `repositorySync` with a semaphore, and abort the sync if you can't get a hold of the semaphore.